### PR TITLE
feat(service): resume PR monitors after worker restart

### PR DESCRIPTION
**Pull request by dimileeh**
Automatically opened by AWF workspace `ws_66d7faf6ef014dea9dcb7939` (agent: `codex`).

### Task
Strict TDD task: implement service-worker recovery for PR monitors after worker/API restarts.

Problem:
The always-on AWF service worker currently dispatches requested workspaces for provisioning and ready workspaces for execution. If the worker process dies or the service is rebuilt while a workspace is already in monitoring_pr, the in-memory PR monitor task disappears and the workspace/PR can be stranded. This is not acceptable for AWF as an always-on agent workspace fabric.

Required behavior:
- On worker run_once/run_forever, detect monitoring_pr workspaces that still need an active PR monitor.
- Resume the existing PR monitor without rerunning setup, agent execution, validation, push, or PR creation.
- Respect max_concurrent_executions and avoid duplicate in-memory monitor tasks for the same workspace.
- Use the workspace's persisted pr_number/pr_url, remote_push_branch, compose_project_name, compose_file_path, resolved profile, agent, auto_merge, and monitor grace settings.
- If a monitoring_pr workspace is missing required persisted PR/branch metadata, fail it with a structured monitor/service failure instead of crashing the worker.
- Do not change completed, failed, cancelled, or destroyed workspaces.
- Keep the implementation small and consistent with existing ControlWorker, WorkspaceExecutor, and service worker factory patterns.

TDD requirements:
- Write failing unit tests first, commit the red tests with the failure snippet in the commit body.
- Add tests that prove monitoring_pr workspaces are resumed after a fresh worker instance starts.
- Add tests that prove ready execution and monitor resume share the configured concurrency limit and do not duplicate tasks.
- Add tests for missing PR metadata failing cleanly.
- Then implement until green.
- Commit with conventional commits. Do not push, open PRs, or switch branches; AWF owns that.

Validation:
uv run --python 3.12 --extra dev ruff check src/awf/control src/awf/service tests/unit/control tests/unit/service
uv run --python 3.12 --extra dev mypy src/awf/control src/awf/service
uv run --python 3.12 --extra dev pytest tests/unit/control tests/unit/service -q

---
Validation: 6 profile command(s) passed inside the workspace container.

### CODE REVIEW

File: tests/unit/control/test_executor_error_paths.py [hunk 1]
```diff
@@ -716,3 +760,107 @@ def _monitor_factory(adapter: Any, profile: Any, workspace: Any) -> _FakeMonitor
...
+        assert monitor_calls == [
+            {
+                "workspace_id": ws_id,
+                "compose_project": "persisted_project",
+                "compose_file": compose_file,
+            }
+        ]
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("field", ["pr_number", "pr_url", "remote_push_branch"])
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply tests/unit/control/test_executor_error_paths.py]: fixed in commit 45cddd1e16609f2ff6eff37e5f7119c50585227b

File: src/awf/control/worker.py [hunk 2]
```diff
@@ -81,13 +82,25 @@ async def run_once(self) -> int:
 
         if self._executor is not None:
             execution_slots = self._available_execution_slots()
-            ready_ids = await self._list_ready(limit=execution_slots)
-            dispatched_ids.update(self._dispatch_ready_executions(ready_ids))
+            if execution_slots > 0:
+                monitoring_ids = await self._list_monitoring_pr(
+                    limit=self._config.max_concurrent_executions
+                )
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/control/worker.py:90]: fixed in commit df7f64b062018dc5e365d1f495eda3abb50e4c89

File: src/awf/control/executor.py [hunk 3]
```diff
@@ -753,8 +753,96 @@ async def _git_in_worktree(args: list[str]):  # type: ignore[no-untyped-def]
...
+    async def resume_pr_monitor(self, workspace_id: str) -> None:
+        """Resume the PR monitor for a workspace already in ``monitoring_pr``.
+
+        This is the service-worker restart path. It intentionally skips setup,
+        agent execution, validation, push, and PR creation; those have already
+        happened before the workspace entered ``monitoring_pr``.
+        """
+        ws = await self._load_workspace(workspace_id)
+        if ws is None or ws.status != WorkspaceStatus.monitoring_pr.value:
+            return
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/control/executor.py:773]: fixed in commit a6671b89eb601fa6ab765998d628d35b478956f1
